src/cache.ts

```diff
@@ -52,30 +54,82 @@ async function buildCacheKeys(): Promise<string[]> {
 }
 
 export async function restoreCache(): Promise<void> {
+  if (!utils.isValidEvent()) {
+    utils.logWarning(
+      `Event Validation Error: The event type ${process.env[Events.Key]} is not supported because it's not tied to a branch or tag ref.`
+    )
+    return
+  }
+
   const startedAt = Date.now()
 
   const keys = await buildCacheKeys()
   const primaryKey = keys.pop()
   const restoreKeys = keys.reverse()
-  core.info(`Primary analysis cache key is '${primaryKey}', restore keys are '${restoreKeys.join(` | `)}'`)
-  process.env[`INPUT_RESTORE-KEYS`] = restoreKeys.join(`\n`)
-  process.env.INPUT_KEY = primaryKey
-
-  process.env.INPUT_PATH = getCacheDirs().join(`\n`)
 
   // Tell golangci-lint to use our cache directory.
   process.env.GOLANGCI_LINT_CACHE = getLintCacheDir()
 
-  await restore()
-  core.info(`Restored cache for golangci-lint from key '${primaryKey}' in ${Date.now() - startedAt}ms`)
+  if (!primaryKey) {
+    utils.logWarning(`Invalid primary key`)
+    return
+  }
+  core.saveState(State.CachePrimaryKey, primaryKey)
+  try {
+    const cacheKey = await cache.restoreCache(getCacheDirs(), primaryKey, restoreKeys)
+    if (!cacheKey) {
+      core.info(`Cache not found for input keys: ${[primaryKey, ...restoreKeys].join(", ")}`)
+      return
+    }
+    // Store the matched cache key
+    utils.setCacheState(cacheKey)
+    const isExactKeyMatch = utils.isExactKeyMatch(primaryKey, cacheKey)
+    utils.setCacheHitOutput(isExactKeyMatch)
```

Comment:
please, remove it: I don't want this to be a part of an exported interface of our GitHub action